migrations/utils.py

```diff
@@ -0,0 +1,167 @@
+class Node:
+    def __init__(self, data):
+        """Initialize a node with data."""
+        self._data = data
+        self._prev_node = None
+        self._next_node = None
+
+    @property
+    def data(self):
+        """Get the data of the node."""
+        return self._data
+
+    @property
+    def prev_node(self):
+        """Get the previous node."""
+        return self._prev_node
+
+    @prev_node.setter
+    def prev_node(self, node):
+        """Set the previous node."""
+        self._prev_node = node
+
+    @property
+    def next_node(self):
+        """Get the next node."""
+        return self._next_node
+
+    @next_node.setter
+    def next_node(self, node):
+        """Set the next node."""
+        self._next_node = node
+
+    def __repr__(self):
+        """Representation of the node."""
+        return f"{self._data}"
+
+    def __eq__(self, other):
+        """Check equality of nodes."""
+        if isinstance(other, Node):
+            return self._data == other.data
+        return False
+
+    def __hash__(self):
+        """Hash the node's data."""
+        return hash(self._data)
+
+
+class LinkedList:
```

Comment:
please add [docstrings](https://peps.python.org/pep-0257/) to all classes.  this one should include something like:

Specialized doubly linked list implementation restricted to unique data values and no cycles.